test/module/shared_model/builders/protobuf/transport_builder_test.cpp

```diff
@@ -250,3 +268,102 @@ TEST_F(TransportBuilderTest, ProposalCreationTest) {
       },
       [](const Error<std::string> &) { FAIL(); });
 }
+
+/**
+ * @given valid proto object of empty block
+ * @when transport builder constructs model object from it
+ * @then original and built objects are equal
+ */
+TEST_F(TransportBuilderTest, EmptyBlockCreationTest) {
+  auto orig_model = createEmptyBlock();
+  testTransport<decltype(orig_model), validation::DefaultEmptyBlockValidator>(
+      orig_model,
+      [&orig_model](const Value<decltype(orig_model)> &model) {
+        ASSERT_EQ(model.value.getTransport().SerializeAsString(),
+                  orig_model.getTransport().SerializeAsString());
+      },
+      [](const Error<std::string> &) { FAIL(); });
+}
+
+/**
+ * @given Valid block protobuf object with no transactions
+ * @when TransportBuilder tries to build BlockVariantType object
+ * @then built object contains EmptyBlock shared model object
+ * AND it is equal to the original object
+ */
+TEST_F(TransportBuilderTest, BlockVariantWithValidEmptyBlock) {
+  auto emptyBlock = createEmptyBlock();
+  interface::BlockVariantType orig_model =
+      std::make_shared<decltype(emptyBlock)>(emptyBlock.getTransport());
+
+  TransportBuilder<interface::BlockVariantType,
+                   validation::DefaultAnyBlockValidator>()
+      .build(emptyBlock.getTransport())
+      .match(
+          [&emptyBlock](const Value<decltype(orig_model)> &model) {
+            iroha::visit_in_place(
+                model.value,
+                [&emptyBlock](
+                    const std::shared_ptr<shared_model::interface::EmptyBlock>
+                        block) { EXPECT_EQ(emptyBlock, *block); },
+                [](const std::shared_ptr<shared_model::interface::Block>) {
+                  FAIL();
+                });
+          },
+          [](const Error<std::string> &) { FAIL(); });
+}
+
+/**
+ * @given Invalid block protobuf object with no transactions
+ * @when TransportBuilder tries to build BlockVariantType object
+ * @then build fails
+ */
+TEST_F(TransportBuilderTest, BlockVariantWithInvalidEmptyBlock) {
+  auto emptyBlock = createInvalidEmptyBlock();
+
+  framework::expected::err(
```

Comment:
This test does not perform any assertions, `err` return `optional` if it matched the error. Please rework according to test description.